src/Pact/Typechecker.hs

```diff
@@ -808,7 +808,7 @@ toFun (TDef td i) = resolveDyn td
 toFun (TDynamic ref mem i) = case (ref, mem) of
   (TVar (Right Var{}) _, TVar (Left (Ref r)) _) -> toFun $ Left <$> r
   _ -> die i "Malformed mod ref"
-toFun t = die (_tInfo t) "Non-var in fun position"
+toFun t = die (_tInfo t) $ "Non-var in fun position " <> show t
```

Comment:
Will remove the call to `show`. This was useful info during debugging, but I opted to re-inline for static TC as opposed to build in the free variable env into the static typechecker.